test/DynamicFees.t.sol

```diff
@@ -149,6 +151,186 @@ contract TestDynamicFees is Test, Deployers, GasSnapshot {
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+
+        vm.expectEmit(true, true, true, true, address(manager));
+        emit Swap(key.toId(), address(swapRouter), -100, 49, 79228162514264333632135824623, 1e18, -1, 500000);
+
+        snapStart("swap with 50 Percent swap fee amountIn");
+        swapRouter.swap(key, params, testSettings, ZERO_BYTES);
+        snapEnd();
+
+        assertEq(_fetchPoolSwapFee(key), 500000);
+    }
+
+    function test_updateDynamicSwapFee_50PercentFee_AmountOut() public {
+        assertEq(_fetchPoolSwapFee(key), 0);
+
+        dynamicFeesHooks.setFee(500000);
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: 100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+
+        vm.expectEmit(true, true, true, true, address(manager));
+        emit Swap(key.toId(), address(swapRouter), -202, 100, 79228162514264329670727698909, 1e18, -1, 500000);
+
+        snapStart("swap with 50 Percent swap fee amountOut");
+        swapRouter.swap(key, params, testSettings, ZERO_BYTES);
+        snapEnd();
+
+        assertEq(_fetchPoolSwapFee(key), 500000);
+    }
+
+    function test_updateDynamicSwapFee_100PercentFee_AmountOut_NoProtocol() public {
+        assertEq(_fetchPoolSwapFee(key), 0);
+
+        dynamicFeesHooks.setFee(1000000);
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: 100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+
+        vm.expectRevert(Pool.CannotSpecifyOutputAmountWithMaxSwapFee.selector);
+        swapRouter.swap(key, params, testSettings, ZERO_BYTES);
+    }
+
+    function test_updateDynamicSwapFee_99PercentFee_AmountOut_WithProtocol() public {
+        assertEq(_fetchPoolSwapFee(key), 0);
+
+        dynamicFeesHooks.setFee(999999);
+
+        vm.prank(address(feeController));
+        manager.setProtocolFee(key, 1000);
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: 100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({takeClaims: false, settleUsingBurn: false});
+
+        vm.expectEmit(true, true, true, true, address(manager));
+        emit Swap(key.toId(), address(swapRouter), -101000000, 100, 79228162514264329670727698909, 1e18, -1, 999999);
+
+        snapStart("swap with dynamic fee and protocol fee");
+        swapRouter.swap(key, params, testSettings, ZERO_BYTES);
+        snapEnd();
+
+        uint256 expectedProtocolFee = uint256(101000000) * 1000 / 1e6;
```

Comment:
where are you getting this value? it should be the delta right? which we return from swap.